src/java/org/apache/cassandra/index/sai/disk/PostingListRangeIterator.java

```diff
@@ -188,9 +188,13 @@ private long getNextRowId() throws IOException
         }
         else
         {
-            segmentRowId = postingList.nextPosting();
+            do
+            {
+                segmentRowId = postingList.nextPosting();
+                // Do not produce a duplicate segment row id.
+            } while (segmentRowId == lastSegmentRowId);
```

Comment:
We probably ought to include a check for `PostingList.END_OF_STREAM` here as well. It may not matter but my concern is that if we don't then `lastSegmentRowId` will get set to it and if this method is called again we'll end up looping forever.